datasets/am/hetq/crawler.py

```diff
@@ -0,0 +1,298 @@
+        context.emit(position)
+        context.emit(occupancy)
+        return person
+    context.log.info(f"No entities emitted for {person_id} ({name_en} / {name_hy})")
+    return None
+
+
+def crawl_list(
+    context: Context,
+    peps: Dict[int, Dict[str, Any]],
+    data: List[Dict[str, Any]],
+    year: int,
+    lang: SupportedLanguage,
+):
+    """Accumulate person info from yearly list.  Do not actually
+    create entities yet because they might not be PEPs."""
+    for entry in data:
+        personID = entry.pop("personID")
+        fullName = entry.pop("fullName")
+        lastPosition = entry.pop("lastPosition")
+        context.audit_data(entry, UNUSED_FIELDS)
+        context.log.debug(
+            f"Found person {personID} ({fullName}, {lastPosition}) in {year}"
+        )
+        if personID not in peps:
+            peps[personID] = {
+                f"name_{lang}": fullName,
+                f"position_{lang}": lastPosition,
+                "years": {year},
+            }
+        else:
+            pep = peps[personID]
+            pep["years"].add(year)
+            if f"name_{lang}" not in pep:
+                pep[f"name_{lang}"] = fullName
+            else:
+                assert pep[f"name_{lang}"] == fullName
+            # NOTE: One might expect there to be different positions
+            # in different years but this is not the case
+            if f"position_{lang}" not in pep:
+                pep[f"position_{lang}"] = lastPosition
+            else:
+                assert pep[f"position_{lang}"] == lastPosition
+
+
+def crawl_relations(
+    context: Context, zipfh: ZipFile, peps: Dict[int, Dict[str, Any]], person_id: int
+):
+    """Read relation graph and create entities."""
+    if person_id not in peps or "entity" not in peps[person_id]:
+        context.log.debug(f"Skipping relations for {person_id}: is not PEP")
+        return
+    # Unnecessary to use the Armenian versions here since the data is the same
+    graph = json.loads(zipfh.read(f"hetq-data/relations/{person_id}-en.json"))
+    if not graph:
+        context.log.warning(
+            f"Graph from hetq-data/relations/{person_id}-en.json is empty"
+        )
+        return
+    # Node IDs *should* be "n{person_id}" but let's map them to be sure
+    node_to_person = {}
+    for n in graph.get("nodes", []):
+        if "id" not in n and "personID" not in n:
+            context.log.warning(
+                f"Strange relation node for {person_id} with no id/personID: {n}"
+            )
+            continue
+        node_to_person[n["id"]] = n["personID"]
+    for e in graph.get("edges", []):
+        source = node_to_person.get(e.get("source"))
+        if source is None:
+            context.log.warning(f"Unknown source node in edge: {e}")
+            continue
+        # Also unnecessary to follow all the edges, only ones exiting
+        # the PEP in question
+        if source != person_id:
+            continue
```

Comment:
not sure if I'm following, but if we have chains of relationships, let's emit them too

Reply:
this dataset might have some similarity to https://www.opensanctions.org/datasets/md_rise_profiles/

Reply:
a person in the dataset, that is

Reply:
Oh, nope. This is not actually the case - people in the relation graph without official positions have a personID but don't actually show up (for obvious reasons in hindsight) in the front-page list of PEPs.  So I will create POI entities for them and also follow their links recursively.

Reply:
Also - they all have profile pages, most of which are empty, but some of which are not (and have DOBs), for instance https://data.hetq.am/hy/profile/577 (Anna Hakobyan)

Reply:
So, there will be a new data dump as unfortunately I did not download these profiles the first time around since they don't show up on the front-page list!